operator/pkg/ingress/ingress.go

```diff
@@ -659,7 +659,8 @@ func (ic *Controller) garbageCollectOwnedResources(ing *slim_networkingv1.Ingres
 		return err
 	}
 
-	if cec != nil {
+	if cec != nil && ic.isEffectiveLoadbalancerModeDedicated(ing) {
```

Comment:
Actually, we could even skip garbage collection completely in case of a shared `Ingress`. This way it becomes obvious that no shared resource is ever garbage collected in these cases. This is also the case for the `Service` & `Endpoints` - in these cases the shared translator only returns `nil` (https://github.com/cilium/cilium/blob/main/operator/pkg/model/translation/translator.go#L90)

Therefore, i would suggest to refactor this to directly skip garbage collection if the `Ingress` is shared.

```
func (ic *Controller) garbageCollectOwnedResources(ing *slim_networkingv1.Ingress) error {
// When the Ingress is in shared mode, shared resources cannot be deleted.
if ic.isEffectiveLoadbalancerModeDedicated(ing) {
  return nil
}
...
```

@sayboras @chaunceyjiang WDYT?

Reply:
+1

Reply:
ah it makes sense with Marco's explanation :+1: